### ISSUE TITLE
Possibly invalid analyzer hint for a missed opportunity to configure the synchronization context

### ISSUE THREAD

**Issue by jcasale** [meziantou/Meziantou.Analyzer#274]
**Context**
Rule identifier: _MA0004_
Target Framework: _.NET 6.0_
Language version: _10.0_

**Describe the bug**
When setting up an instance of the `Microsoft.Data.SqlClient`, I noticed the analyzer hints about a missed opportunity to configure the synchronization context. Since configuring the synchronization context changes the return type, I am not clear whether the warning is helpful in this case.

**To Reproduce**
The following excerpt provokes the warning:

````c#
using Microsoft.Data.SqlClient;

await using var connection = new SqlConnection(" ... connection string ... ");
````

![SynchronizationContextError](https://user-images.githubusercontent.com/9957114/143784098-9faf4c4e-4116-43fe-8176-8833bbbd8d32.png)

**Comment by meziantou** [comment]
The diagnostic is ok. You can fix the code by using:

````c#
var connection = new SqlConnection(" ... connection string ... ");
await using var file = connection.ConfigureAwait(false);
````

I agree this is not the easiest pattern, but if you don't want to use the `SynchronizationContext` you have to use it or to call `DisposeAsync` manually.

Here's the code generated by the compiler for both code:

````c#
var connection = new SqlConnection(" ... connection string ... ");
try { }
finally
{
    await connection.DisposeAsync();
}
````

Using `ConfigureAwait`

````c#
var connection = new SqlConnection(" ... connection string ... ");
try { }
finally
{
    await connection.DisposeAsync().ConfigureAwait(false);
}
````

**Comment by jcasale** [comment]
It's unfortunate the C# 8.0 alternative syntax for the using statement doesn't allow for the variable to be omitted like the old syntax.

```cs
var connection = new SqlConnection(" ... connection string ... ");
await using var file = connection.ConfigureAwait(false);
```
Without the variable, it must become the following which leads back to the layered indenting:
```cs
var connection = new SqlConnection(" ... connection string ... ");
await using (connection.ConfigureAwait(false))
{
}
```

Thanks for the quick reply and guidance.

**Comment by meziantou** [comment]
Feel free to upvote the issue in csharplang: https://github.com/dotnet/csharplang/issues/2235
